ENH: rudimentary ls and validate commands

yarikoptic opened the pull request:
TODOs (for the future, not PR):
- Proper tests
- Extend interface
- Add recursive files discovery
- More metadata fields to be listed (and provide option to select which ones)

PS Sits on top of #7

danlamanna commented on dandi/cli/command.py:
```diff
@@ -0,0 +1,223 @@
+#
+
+@main.command()
+@click.argument('paths', nargs=-1, type=click.Path())
```
Should this be `click.Path(exists=True, dir_okay=False)` ?

yarikoptic replied:
I think so (at least for now, until I add directories treatment) - I will change it now. 
As you can feel I am not a `click` expert (yet). Thanks!

danlamanna commented on dandi/cli/command.py:
```diff
@@ -0,0 +1,223 @@
+        raise SystemExit(1)
+    else:
+        print(
+            crayons.green("No validation errors among {} files".format(len(files)),
```
I'm unfamiliar with crayons, does it provide something more than `click.style` http://click.palletsprojects.com/en/7.x/utils/#ansi-colors?

yarikoptic replied:
oh, I didn't know about that feature of click.  most likely it would be sufficient - I will check. thanks!
I picked up crayons idea from pipenv which also uses click.

kyleam commented on dandi/cli/command.py:
```diff
@@ -0,0 +1,223 @@
+                #lgr.error("File is not available: %s", exc)
+                pass
+            out(rec)
+        out.wait()
```
You're using the context manager.  Its `__exit__` already calls `wait`, so you don't need this here (or if you do, it's a bug).

yarikoptic replied:
got it.  There was indeed some oddity in `dandi ls` quitting right away (either not waiting for the callback at all or just not rendering its output), so I added the `.wait()`, but I think it solved nothing, I will strip it away. thanks

yarikoptic commented:
ok, I have addressed the comments -- thanks for the feedback!
it is ugly and untested but seems to work.  I will merge it but if you see anything in new code worth immediate action, either leave a comment here or add a new issue/PR.
